app/controllers/events_controller.rb

```diff
@@ -8,8 +8,9 @@ def index
     events << Course.past.all
     events << Meeting.past.all
     events << Event.past.all
```

Comment:
If the limit is 40 events, you can afford to limit these three calls to fetching the last 40 results for `Course`/`Meeting`/`Event`.